mod.go

```diff
@@ -11,18 +11,28 @@ import (
 )
 
 // hashDir is like dirhash.HashDir except that it ignores the
-// gohack hash file in the top level directory.
-func hashDir(dir string) (string, error) {
+// gohack hash file in the top level directory, and auto-generated
+// go.mod files.
+func hashDir(dir string, modulePath string) (string, error) {
 	files, err := dirhash.DirFiles(dir, "")
 	if err != nil {
 		return "", err
 	}
 	j := 0
 	for _, f := range files {
-		if f != hashFile {
-			files[j] = f
-			j++
+		if f == hashFile {
+			continue
+		} else if f == "go.mod" {
```

Comment:
I don't think there is a scenario in which a `go.sum` file will appear, but worth guarding against this too?

Reply:
I don't think this is an issue here, as gohack doesn't create go.mod files (it creates a go.mod file when the module doesn't have one, so that it can be treated as a valid module). An open question is whether we should add requirements to the auto-generated go.mod file so that the module can be built without modifying it. If we did that, we'd probably want to exclude go.sum from the checksum too, I guess.